src/poggit/release/submit/SubmitModule.php

```diff
@@ -0,0 +1,894 @@
+                <br/>
+                If you only rewrote the plugin but did not take any code from the original author, consider using a new
+                plugin name, or at least add something like <code>_New</code> behind the plugin name. Consider adding
+                the original author as a <em>Requester</em> in the <em>Producers</em> field below.<br/>
+                If you have used some code from the original author but have made major changes to the plugin, you are
+                allowed to submit this plugin from your <em>fork</em> repo, but you <strong>must</strong> add the
+                original author as a <em>collaborator</em> in the <em>Producers</em> field below.
+            </p>
+            <p class="remark">Note: If you don't submit this form within three hours after loading this page, this form
+                will become invalid and you will have to reload this page.</p>
+            <div class="form-table">
+                <h3>Loading...</h3>
+                <p>If this page doesn't load in a few seconds, try refreshing the page. You must enable JavaScript to
+                    use
+                    this page.</p>
+            </div>
+        </div>
+        <?php $this->bodyFooter(); ?>
+        <?php $this->includeJs("newSubmit"); ?>
+        </body>
+        </html>
+        <?php
+        OutputManager::endMinifyHtml($minifier);
+    }
+
+    private function prepareAssocData() {
+        if($this->mode === self::MODE_UPDATE) {
+            $this->assocChildren = [];
+            foreach(Mysql::query("SELECT releaseId, name, version FROM releases WHERE parent_releaseId = ? AND state >= ?",
+                "ii", $this->refRelease->releaseId, Release::STATE_SUBMITTED) as $row) {
+                $this->assocChildren[(int) $row["releaseId"]] = $child = new stdClass();
+                $child->name = $row["name"];
+                $child->description = $row["version"];
+            }
+        }
+    }
+
+    private function loadPoggitYml() {
+        try {
+            $response = Curl::ghApiGet("repositories/{$this->repoInfo->id}/contents/.poggit.yml?ref=" . $this->buildInfo->sha, Session::getInstance()
+                ->getAccessToken(), ["Accept: application/vnd.github.VERSION.raw"], true);
+        } catch(GitHubAPIException $e) {
+            Meta::getLog()->jwtf($e);
+            $this->errorBadRequest(".poggit.yml missing");
+            return;
+        }
+        try {
+            $data = yaml_parse($response);
+            if(!is_array($data)) {
+                throw new \Exception("Error parsing .poggit.yml");
+            }
+        } catch(NativeError $e) {
+            Meta::getLog()->jwtf($e);
+            $this->errorBadRequest("Error parsing .poggit.yml");
+            return;
+        }
+
+        $this->poggitYml = $data;
+        $this->poggitYmlProject = array_change_key_case($data["projects"], CASE_LOWER)[strtolower($this->buildProjectName)] ?? null;
```

Comment:
On opening existing projects (with hacky fixes for the above errors): Undefined index: projects in /Users/jazzwhistle/Poggit/poggit/src/poggit/release/submit/SubmitModule.php#782